params/protocol_params.go

```diff
@@ -95,9 +95,11 @@ const (
 	// TODO: make this cost variable- https://github.com/celo-org/geth/issues/250
 	FractionMulExpGas uint64 = 1050 // Cost of performing multiplication and exponentiation of fractions to an exponent of up to 10^3.
 	// TODO(kobigurk):  Figure out what the actual gas cost of this contract should be.
-	ProofOfPossessionGas uint64 = 50000 // Cost of verifying a BLS proof of possession.
-	GetValidatorGas      uint64 = 5000  // Cost of reading a validator's address.
-	GetEpochSizeGas      uint64 = 1000  // Cost of querying the number of blocks in an epoch.
+	ProofOfPossessionGas        uint64 = 50000 // Cost of verifying a BLS proof of possession.
+	GetValidatorGas             uint64 = 5000  // Cost of reading a validator's address.
+	GetEpochSizeGas             uint64 = 1000  // Cost of querying the number of blocks in an epoch.
+	GetBlockNumberFromHeaderGas uint64 = 10000 // Cost of decoding a block header.
```

Comment:
How were these decided?

Reply:
I used the GetEpochSizeGas = 1000 as a rule of thumb, and said that if (returning a constant) costs x, then parsing a header costs 10x, and parsing then hashing a header is 20x.  But now that you mention it, it's not consistent with Sha256BaseGas and Sha256PerWordGas. Which would indicate hashing a header of 508 bytes costs only 252 gas, not 10,000.